lang/src/org/partiql/lang/eval/EvaluatingCompiler.kt

```diff
@@ -743,6 +747,13 @@ internal class EvaluatingCompiler(
             // NOTE: we cannot use a [HashSet<>] here because [ExprValue] does not implement [Object.hashCode] or
             // [Object.equals].
             val precomputedLiteralsMap = values
+                .asSequence()
+                .map { value ->
+                    when (value) {
+                        is PartiqlAst.Expr.Struct -> if (value.isSingleValueOptimizedStruct()) value.fields[0].second else value
+                        else -> value
+                    }
+                }
```

Comment:
What's the use-case for this optimization?  Has this been reported by a customer as being slow?  If not, I would avoid optimizing this as it may be premature.  The other optimizations here were added as a direct result of customer feedback.

Reply:
ACK. I can modify. Thanks!